addon/wait-for-coroutine.ts

```diff
@@ -0,0 +1,74 @@
+import TestWaiter from './test-waiter';
+
+const COROUTINE_WAITER = new TestWaiter<Generator>('coroutine-waiter');
+
+/**
+ * A convenient utility function to simplify waiting for a coroutine in a
+ * context such as ember-concurrency. Can be used in both decorator and
+ * function form.
+ *
+ * @public
+ * @param coroutine {Generator<T, TReturn, TNext>} The coroutine to track async operations for
```

Comment:
Can we make the jsdoc params match the signature? Specifically, the `coroutine` param is actually `target` below, no?